services/continuous_querier/service_test.go

```diff
@@ -92,6 +93,84 @@ func TestContinuousQueryService_Run(t *testing.T) {
 	s.Close()
 }
 
+func TestContinuousQueryService_ResampleOptions(t *testing.T) {
+	s := NewTestService(t)
+	ms := NewMetaStore(t)
+	ms.CreateDatabase("db", "")
+	ms.CreateContinuousQuery("db", "cq", `CREATE CONTINUOUS QUERY cq ON db RESAMPLE EVERY 10s FOR 2m BEGIN SELECT mean(value) INTO cpu_mean FROM cpu GROUP BY time(1m) END`)
+	s.MetaStore = ms
+
+	db, err := s.MetaStore.Database("db")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cq, err := NewContinuousQuery(db.Name, &db.ContinuousQueries[0])
+	if err != nil {
+		t.Fatal(err)
+	} else if cq.Resample.Every != 10*time.Second {
+		t.Errorf("expected resample every to be 10s, got %s", influxql.FormatDuration(cq.Resample.Every))
+	} else if cq.Resample.For != 2*time.Minute {
+		t.Errorf("expected resample for 2m, got %s", influxql.FormatDuration(cq.Resample.For))
+	}
+
+	// Set RunInterval high so we can trigger using Run method.
+	s.RunInterval = 10 * time.Minute
+
+	done := make(chan struct{})
+	expectCallCnt := 0
+	callCnt := 0
```

Comment:
I copied this pattern from earlier in the file, so it might be a good idea to go over this file again at some point and get this file using the same conventions as the rest of the codebase.